src/tfr_navigation/src/navigator.cpp

```diff
@@ -0,0 +1,135 @@
+#include <navigator.h>
+
+/**
+ *  constructs the sever and binds it to it's execution callback
+ *  displays set parameters and warnings to the user
+ * */
+Navigator::Navigator(ros::NodeHandle &n,std::string name) : node{n},  server{n, name,
+    boost::bind(&Navigator::navigate, this, _1) ,false}, action_name{name},
+    goal_manager()
+{
+    ROS_DEBUG("Navigation server constructed %f", ros::Time::now().toSec());
+    //get parameters
+    ros::param::param<std::string>("~odometry_topic", odometry_topic,
+            "/fused_odom");
+    ros::param::param<float>("~rate", rate, 1);
+    ros::param::param<std::string>("~frame_id", frame_id, "base_footprint");
+
+    odom_subscriber = node.subscribe(odometry_topic, 5,
+            &Navigator::update_position, this);
+
+    //display parameters to the user
+    ROS_DEBUG(" name:           %s", action_name.c_str());
+    ROS_DEBUG(" frame_id:       %s", frame_id.c_str());
+    ROS_DEBUG(" odometry_topic: %s", odometry_topic.c_str());
+    ROS_DEBUG(" rate:           %f", rate);
+
+
+    server.start();
+
+    ROS_INFO("Navigation server awaiting connection");
+}
+
+/**
+ *  Goal: 
+ *      -uint8_t code corresponding to where we want to navigate. Goal list is
+ *      described in Navigation.action in the tfr_msgs package
+ *  Feedback:
+ *      -uint8_t code corresponding to our current status described in 
+ *      Navigation.action in the tfr_msgs package
+ *      -Pose describing the current position
+ *      -Pose describing the final targeted position
+ *  Response
+ *      -uint8_t code corresponding to the finsal status described in 
+ *      Navigation.action in the tfr_msgs package
+ *      -Pose describing our final position
+ *  NOTE NOT THREAD SAFE
+ * */
+void Navigator::navigate(const tfr_msgs::NavigationGoalConstPtr &goal)
+{
+    ROS_INFO("Navigation server started");
+
+    //start with initial goal
+    goal_manager.location_code = (*goal).location_code;
```

Comment:
You could also use `goal->location_code` here, which might be easier to read.

Reply:
oh sure I was silly

Reply:
Done!